File: 1001-1100/1024-Video Stitching/1024-Video Stitching.py
class Solution:
    def videoStitching(self, clips: List[List[int]], T: int) -> int:
        clips.sort()
        maxT = count = i = 0
        while i < len(clips):
            start, maxEnd = clips[i]
            if start > maxT:
                return -1
            while i < len(clips) and clips[i][0] <= maxT:
                maxEnd = max(maxEnd, clips[i][1])
                i += 1
            count += 1
            if maxEnd >= T:
                return count
            maxT = maxEnd
        return -1

File: 1001-1100/1024-Video Stitching/1024-Video Stitching.cpp
class Solution {
public:
    int videoStitching(vector<vector<int>>& clips, int T) {
        sort(clips.begin(), clips.end());
        int maxT = 0, count = 0, i = 0;
        while (i < clips.size()) {
            int maxEnd = clips[i][1];
            if (clips[i][0] > maxT) {
                return -1;
            }
            while (i < clips.size() && clips[i][0] <= maxT) {
                maxEnd = max(maxEnd, clips[i++][1]);
            }
            ++count;
            if (maxEnd >= T) {
                return count;
            }
            maxT = maxEnd;
        }
        return -1;
    }
};
